File: sem4/mpp/mpp-proiect-csharp-vreauladudu/persistence/repository/database/UserDatabase.cs
using mpp_proiect_csharp_vreauladudu.domain;
using System;
using System.Data.SQLite;

namespace mpp_proiect_csharp_vreauladudu.repository.database
{
    public class UserDatabase : AbstractDatabase<int, User>, UserRepository
    {
        public UserDatabase() : base("Users") { }

        protected override User extractEntity(SQLiteDataReader result)
        {   
            int id = Convert.ToInt32(result["id"]);
            String username = Convert.ToString(result["username"]);
            String password = Convert.ToString(result["password"]);

            User user = new User();
            user.Id = id;
            user.Username = username;
            user.Password = password;
            return user;
        }

        protected override SQLiteCommand findStatement(int id)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "SELECT * FROM Users WHERE [id] = @id";
            command.Parameters.Add(new SQLiteParameter("@id", id));
            return command;
        }

        protected override SQLiteCommand saveStatement(User user)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "INSERT INTO Users(id, username, password) VALUES (@id, @username, @password)";
            command.Parameters.Add(new SQLiteParameter("@id", user.Id));
            command.Parameters.Add(new SQLiteParameter("@username", user.Username));
            command.Parameters.Add(new SQLiteParameter("@password", user.Password));
            return command;
        }

        protected override SQLiteCommand deleteStatement(int id)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "DELETE FROM Users WHERE [id] = @id";
            command.Parameters.Add(new SQLiteParameter("@id", id));
            return command;
        }

        protected override SQLiteCommand updateStatement(User user)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "UPDATE Users SET [username] = @username, [password] = @password WHERE [id] = @id";
            command.Parameters.Add(new SQLiteParameter("@username", user.Username));
            command.Parameters.Add(new SQLiteParameter("@password", user.Password));
            command.Parameters.Add(new SQLiteParameter("@id", user.Id));
            return command;
        }

        public User isValidUser(String username, String password)
        {
            using (SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand())
            {
                command.CommandText = "SELECT * FROM Users WHERE [username] = @username AND [password] = @password";
                command.Parameters.Add(new SQLiteParameter("@username", username));
                command.Parameters.Add(new SQLiteParameter("@password", password));

                using (SQLiteDataReader result = command.ExecuteReader())
                {
                    return result.Read() ? this.extractEntity(result) : null;
                }
            }
        }
    }
}

File: sem4/mpp/mpp-proiect-java-vreauladudu/persistence/src/main/java/moto/repository/database/UserDatabase.java
package moto.repository.database;

import moto.User;
import moto.repository.UserRepository;

import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

public class UserDatabase extends AbstractDatabase<Integer, User> implements UserRepository
{
    public UserDatabase(Properties properties)
    {
        super(properties, "Users");
    }

    @Override
    protected User extractEntity(ResultSet resultSet) throws SQLException
    {
        Integer id = resultSet.getInt("id");
        String username = resultSet.getString("username");
        String password = resultSet.getString("password");

        User user = new User();
        user.setId(id);
        user.setUsername(username);
        user.setPassword(password);
        return user;
    }

    @Override
    protected PreparedStatement findStatement(Integer id) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("SELECT * from Users WHERE id = ?");
        statement.setInt(1, id);
        return statement;
    }

    @Override
    protected PreparedStatement saveStatement(User user) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("INSERT INTO Users(id, username, password) VALUES (?, ?, ?)");
        statement.setInt(1, user.getId());
        statement.setString(2, user.getUsername());
        statement.setString(3, user.getPassword());
        return statement;
    }

    @Override
    protected PreparedStatement deleteStatement(Integer id) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("DELETE FROM Users WHERE id = ?");
        statement.setInt(1, id);
        return statement;
    }

    @Override
    protected PreparedStatement updateStatement(User user) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("UPDATE Users SET username = ?, password = ? WHERE id = ?");
        statement.setString(1, user.getUsername());
        statement.setString(2, user.getPassword());
        statement.setInt(3, user.getId());
        return statement;
    }

    @Override
    public User isValidUser(String username, String password) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("SELECT * FROM Users WHERE username = ? AND password = ?");
        statement.setString(1, username);
        statement.setString(2, password);
        ResultSet resultSet = statement.executeQuery();
        return resultSet.next() ? this.extractEntity(resultSet) : null;
    }
}